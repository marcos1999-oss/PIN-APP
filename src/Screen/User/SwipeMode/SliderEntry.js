import React, { Component } from "react";
import { View, Text, Image, TouchableOpacity } from "react-native";
import { ParallaxImage } from "react-native-snap-carousel";
import numeral from 'numeral'
import { isUndefined, isNull } from 'lodash'
import styles from "./SliderEntry.style";
import CustomIcon from "../../../CustomIcon";
import { FeatureList } from '../../../Components/Common'
import { CachedImage } from 'react-native-cached-images';

export default class SliderEntry extends Component {
  constructor(props) {
    super(props);
  }

  get image() {
    const {
      data: { photo },
      parallax,
      parallaxProps,
      even,
    } = this.props;

    return parallax ? (
      <ParallaxImage
        source={{ uri: photo.url }}
        containerStyle={[
          styles.imageContainer,
          even ? styles.imageContainerEven : {},
        ]}
        style={styles.image}
        parallaxFactor={0.35}
        showSpinner={true}
        spinnerColor={even ? "rgba(255, 255, 255, 0.4)" : "rgba(0, 0, 0, 0.25)"}
        {...parallaxProps}
      />
    ) : (
        <CachedImage
          source={{ uri: photo.url }}
          style={styles.image}
        />
      );
  };

  render() {
    const {
      data: { title, likesCount, viewsCount, firm },
    } = this.props;

    return (
      <TouchableOpacity
        activeOpacity={1}
        style={styles.slideInnerContainer}
        onPress={() => {
          this.props.onPressItem(this.props.data);
        }}
      >
        {/* <View style={styles.shadow} /> */}
        <View style={styles.imageContainer}>
          {this.image}
          <View style={styles.radiusMask} />
        </View>
        <View style={styles.dealInfoContainer}>
          <View style={styles.dealLogoContainer}>
            <View style={styles.dealLogoWrapper}>
              <CachedImage source={{ uri: firm.photo.url }} style={styles.dealLogo} />
            </View>
            <Text numberOfLines={1} style={styles.dealBranch}>{firm.name}</Text>
          </View>
          <View style={styles.dealTextContainer}>
            <Text style={styles.subtitle} numberOfLines={1}>
              {title}
            </Text>

            <FeatureList
              features={firm.features}
              containerStyle={styles.features}
              style={{ padding: 4 }}
              imageStyle={styles.featureIcons}
            />

            <View style={styles.dealPerformanceContainer}>
              <View style={styles.likeContainer}>
                <CustomIcon name="heart" style={styles.likeIcon} />
                <Text style={styles.performanceText}>{numeral(likesCount).format('0,0')} Likes</Text>
              </View>
              <View style={styles.watchContainer}>
                <CustomIcon name="eye" />
                <Text style={styles.performanceText}>{numeral(viewsCount).format('0,0')} Views </Text>
              </View>
            </View>
          </View>
        </View>
      </TouchableOpacity>
    );
  }
}
