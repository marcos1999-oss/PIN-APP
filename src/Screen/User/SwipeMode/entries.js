export const ENTRIES1 = [
  {
    title: "Beautiful and dramatic Antelope Canyon",
    subtitle: "Jumbo Deal $2.99 Get 2 tacos, Jumbo jack,  48 char",
    illustration: "https://i.imgur.com/KZsmUi2l.jpg",
    // illustration: require('../../../../assets/img/deal-1.png')
  },
  {
    title: "Earlier this morning, NYC",
    subtitle: "Jumbo Deal $2.99 Get 2 tacos, Jumbo jack,  48 char",
    illustration: "https://i.imgur.com/KZsmUi2l.jpg",
    // illustration: require('../../../../assets/img/deal-2.png')
  },
  {
    title: "White Pocket Sunset",
    subtitle: "Lorem ipsum dolor sit amet et nuncat ",
    illustration: "https://i.imgur.com/KZsmUi2l.jpg",
    // illustration: require('../../../../assets/img/deal-3.png')
  },
  {
    title: "Acrocorinth, Greece",
    subtitle: "Lorem ipsum dolor sit amet et nuncat mergitur",
    illustration: "https://i.imgur.com/KZsmUi2l.jpg",
  },
  {
    title: "The lone tree, majestic landscape of New Zealand",
    subtitle: "Lorem ipsum dolor sit amet",
    illustration: "https://i.imgur.com/2nCt3Sbl.jpg",
  },
  {
    title: "Middle Earth, Germany",
    subtitle: "Lorem ipsum dolor sit amet",
    illustration: "https://i.imgur.com/lceHsT6l.jpg",
  },
];
