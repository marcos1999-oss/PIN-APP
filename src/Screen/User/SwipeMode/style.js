import { StyleSheet } from "react-native";
import { verticalScale, scale, moderateScale } from "../../../Utils/scaling";

export const colors = {
  black: "#1a1917",
  gray: "#888888",
  background1: "#B721FF",
  background2: "#21D4FD",
};

export const endedIconSize = verticalScale(100);

export default StyleSheet.create({
  endedContainer: {
    flex: 1,
    alignItems: "center",
    flexDirection: "column",
    paddingTop: verticalScale(120),
  },
  endedText: {
    marginTop: verticalScale(20),
    fontFamily: "Helvetica",
    fontSize: moderateScale(14),
    textAlign: "center",
  },
  swiperContainer: {
    position: "relative",
    backgroundColor: "transparent",
    height: verticalScale(393.41),
    paddingHorizontal: scale(35),
    marginTop: verticalScale(15),
    alignItems: "center",
    justifyContent: "center",
  },
  cardContainer: {
    // position: 'relative',
    top: 0,
    left: scale(20),
    width: "100%",
    height: "100%",
  },
  //   card: {
  //     // flex: 1,
  //     height: '90%',
  //     // position: 'relative',
  //     borderRadius: 4,
  //     borderWidth: 2,
  //     borderColor: '#E8E8E8',
  //     justifyContent: 'center',
  //     backgroundColor: 'white',
  //   },
  dislikeLabel: {
    backgroundColor: "transparent",
    borderColor: "#F40100",
    color: "#F40100",
    borderWidth: verticalScale(4),
  },
  likeLabel: {
    backgroundColor: "transparent",
    borderColor: "#3FC94F",
    color: "#3FC94F",
    borderWidth: verticalScale(4),
  },
  safeArea: {
    flex: 1,
    backgroundColor: colors.black,
  },
  container: {
    flex: 1,
    backgroundColor: colors.background1,
  },
  gradient: {
    ...StyleSheet.absoluteFillObject,
  },
  scrollview: {
    flex: 1,
  },
  exampleContainer: {
    // paddingVertical: verticalScale(20)
  },
  exampleContainerLight: {
    // backgroundColor: 'white'
  },
  title: {
    paddingHorizontal: 30,
    backgroundColor: "transparent",
    color: "rgba(255, 255, 255, 0.9)",
    fontSize: 20,
    fontWeight: "bold",
    textAlign: "center",
  },
  titleDark: {
    color: colors.black,
  },
  subtitle: {
    marginTop: 5,
    paddingHorizontal: 30,
    backgroundColor: "transparent",
    color: "rgba(255, 255, 255, 0.75)",
    fontSize: 13,
    fontStyle: "italic",
    textAlign: "center",
  },
  slider: {
    marginTop: verticalScale(15),
    overflow: "visible", // for custom animations
  },
  sliderContentContainer: {
    paddingVertical: scale(10), // for custom animation
  },
  paginationContainer: {
    paddingVertical: scale(8),
  },
  paginationDot: {
    width: scale(8),
    height: verticalScale(8),
    borderRadius: verticalScale(4),
    marginHorizontal: scale(8),
  },
  itemContainer: {
    flexDirection: "column",
    height: verticalScale(390),
    borderRadius: verticalScale(20),
    overflow: "hidden",
  },
  imageContainer: {
    height: verticalScale(325),
    // borderTopLeftRadius: 20,
    // borderTopRightRadius: 20,
    backgroundColor: "green",
  },
  buttonGroup: {
    marginTop: verticalScale(40),
    height: verticalScale(50),
    width: "100%",
    flexDirection: "row",
    justifyContent: "space-around",
    alignSelf: "flex-end",
  },
  actionButton: {
    height: "100%",
    width: scale(150),
    borderRadius: verticalScale(20),
    justifyContent: "center",
    shadowColor: "#000",
    shadowOffset: { width: verticalScale(3), height: verticalScale(3) },
    shadowOpacity: 0.3,
    shadowRadius: 2,
    justifyContent: "center",
    alignItems: "center",
  },
  actionText: {
    textTransform: "uppercase",
    fontSize: moderateScale(16),
    color: "white",
    fontWeight: "bold",
  },
  distanceCircle: {
    position: "absolute",
    zIndex: 999,
    right: scale(-5),
    bottom: verticalScale(-15),
    backgroundColor: "#5F92F3",
    width: verticalScale(35),
    minWidth: verticalScale(35),
    height: verticalScale(35),
    minHeight: verticalScale(35),
    borderRadius: verticalScale(35) / 2,
    borderWidth: 3,
    borderColor: "white",
    justifyContent: "center",
    alignItems: "center",
  },
  distanceText: {
    fontSize: moderateScale(9),
    color: "white",
    fontWeight: "bold",
  },
});
