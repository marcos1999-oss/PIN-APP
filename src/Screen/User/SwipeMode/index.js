import React, { Component } from "react";
import { View, Text, SafeAreaView, TouchableHighlight } from "react-native";
import Swiper from "react-native-deck-swiper";
import { isUndefined, isEmpty } from 'lodash';
import { showMessage } from "react-native-flash-message";
import SliderEntry from "./SliderEntry";
import styles from "./style";
import NoDealsView from "../NoDeals";
//Redux
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {
  getPostsPagination,
} from '../../../redux/selectors/index';
import * as postActions from '../../../redux/actions/postActions';

class SwipeMode extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isEmpty: false,
      likePressed: false,
      dislikePressed: false,
    };
  }

  componentWillReceiveProps(nextProps) {
    if (!isEmpty(nextProps.posts) && isEmpty(this.props.post)) {
      const post = nextProps.posts[nextProps.initialCardIndex];

      this.props.postActions.viewPost({ post });
    }
  }

  likePost = () => {
    const { post } = this.props;

    this.props.postActions.likePost({
      post,
      onSuccess: () => {
        this.swiper.swipeRight();
      },
      onFail: (error) => {
        showMessage({
          message: error,
          type: 'danger',
        });
      },
    });
  };

  dislikePost = () => {
    const { post } = this.props;

    this.props.postActions.dislikePost({
      post,
      onSuccess: () => {
        this.swiper.swipeLeft();
      },
      onFail: (error) => {
        showMessage({
          message: error,
          type: 'danger',
        });
      },
    });
  };

  onPressDeal = item => {
    this.props.postActions.viewPost({
      post: item,
      callback: () => {
        this.props.navigation.navigate('UserAdView', { isUser: true })
      },
    });
  };

  getDistance = () => {
    return `${this.distance()} mi`;
  };

  _renderCard = (card, _index) => {
    if (isUndefined(card)) {
      return (<View></View>);
    }

    return <SliderEntry data={card} onPressItem={this.onPressDeal} />;
  };

  renderEmpty = () => {
    return (<NoDealsView />)
  };

  _onHideUnderlay = (isLiked) => {
    if (isLiked) this.setState({ likePressed: false })
    else this.setState({ dislikePressed: false });
  }
  _onShowUnderlay = (isLiked) => {
    if (isLiked) this.setState({ likePressed: true })
    else this.setState({ dislikePressed: true });
  }

  distance = () => {
    if (this.props.me !== undefined && this.props.post !== undefined && this.props.post.firm !== undefined) {
      const lat1 = this.props.me.latitude;
      const lon1 = this.props.me.longitude;
      const lat2 = this.props.post.firm.lat;
      const lon2 = this.props.post.firm.lng;
      const R = 6371; // km (change this constant to get miles)
      const dLat = (lat2 - lat1) * Math.PI / 180;
      const dLon = (lon2 - lon1) * Math.PI / 180;
      const a = Math.sin(dLat / 2) * Math.sin(dLat / 2) +
        Math.cos(lat1 * Math.PI / 180) * Math.cos(lat2 * Math.PI / 180) *
        Math.sin(dLon / 2) * Math.sin(dLon / 2);
      const c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
      const d = (R * c) / 1.60934;
      if (d < 1) return Math.round(d.toFixed(2) * 10) / 10;
      else return Math.round(d);
    }
  }

  handlePagination() {
    if (this.props.pagination.currentPage !== this.props.pagination.total) {
      this.setState({ isEmpty: false });
      this.props.postActions.fetchPosts({
        currentPage: this.props.pagination.currentPage + 1,
        perPage: this.props.pagination.perPage,
      });
    } else {
      this.setState({ isEmpty: true });
    }
  }

  render() {
    if (this.state.isEmpty) {
      return this.renderEmpty();
    } else if (this.props.postsMeta.loading) {
      return this.renderEmpty();
    } else if (this.props.data.length === 0) {
      return this.renderEmpty();
    }
    const { likePressed, dislikePressed } = this.state;
    return (
      <View>
        <Swiper
          cards={this.props.data}
          containerStyle={styles.swiperContainer}
          cardStyle={styles.cardContainer}
          ref={swiper => this.swiper = swiper}
          renderCard={this._renderCard}
          onSwiped={cardIndex => {
            const post = this.props.posts[cardIndex + 1];

            this.props.postActions.viewPost({ post });
            this.props.postActions.swipePost({
              post,
              onSuccess: () => { },
              onFail: (error) => {
                showMessage({
                  message: error,
                  type: 'danger',
                });
              },
            });
          }}
          onSwipedAll={() => this.handlePagination()}
          cardIndex={this.props.initialCardIndex}
          disableBottomSwipe={true}
          disableTopSwipe={true}
          stackSize={3}
          stackSeparation={15}
          overlayLabels={{
            left: {
              title: "DISLIKE",
              style: {
                label: {
                  ...styles.dislikeLabel,
                },
                wrapper: {
                  flexDirection: "column",
                  alignItems: "flex-end",
                  justifyContent: "flex-start",
                  marginTop: 30,
                  marginLeft: -30,
                },
              },
            },
            right: {
              title: "LIKE",
              style: {
                label: {
                  ...styles.likeLabel,
                },
                wrapper: {
                  flexDirection: "column",
                  alignItems: "flex-start",
                  justifyContent: "flex-start",
                  marginTop: 30,
                  marginLeft: 30,
                },
              },
            },
          }}
          animateOverlayLabelsOpacity
          animateCardOpacity
        // swipeBackCard
        />

        {!isEmpty(this.props.post) &&
          <SafeAreaView style={styles.buttonGroup}>
            <View>
              <TouchableHighlight
                style={{ ...styles.actionButton, backgroundColor: "#F40100", shadowOpacity: dislikePressed ? 0 : 0.5 }}
                onHideUnderlay={() => this._onHideUnderlay(false)}
                onShowUnderlay={() => this._onShowUnderlay(false)}
                underlayColor={"#F07F7F"}
                onPress={this.dislikePost.bind(this)}
              >
                <Text style={styles.actionText}>DISLIKE</Text>
              </TouchableHighlight>
            </View>

            <View>
              <View>
                <TouchableHighlight
                  style={{ ...styles.actionButton, backgroundColor: "#5F92F3", shadowOpacity: likePressed ? 0 : 0.5 }}
                  onHideUnderlay={() => this._onHideUnderlay(true)}
                  onShowUnderlay={() => this._onShowUnderlay(true)}
                  underlayColor={"#5F92F3"}
                  onPress={this.likePost.bind(this)}
                >
                  <Text style={styles.actionText}>LIKE</Text>
                </TouchableHighlight>
              </View>

              <View style={styles.distanceCircle}>
                <Text style={styles.distanceText}>{this.getDistance()}</Text>
              </View>
            </View>
          </SafeAreaView>
        }
      </View>
    );
  }
}

const mapStateToProps = state => ({
  pagination: getPostsPagination(state),
});

const mapDispatchToProps = dispatch => ({
  postActions: bindActionCreators(postActions, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(SwipeMode)
