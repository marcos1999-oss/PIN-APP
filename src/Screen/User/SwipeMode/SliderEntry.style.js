import { StyleSheet, Dimensions, Platform } from "react-native";
import { colors } from "./style";
import { verticalScale, moderateScale, scale } from "../../../Utils/scaling";

const IS_IOS = Platform.OS === "ios";
const { width: viewportWidth, height: viewportHeight } = Dimensions.get(
  "window"
);

function wp(percentage) {
  const value = (percentage * viewportWidth) / 100;
  return Math.round(value);
}

// const slideHeight = viewportHeight * 0.36;
const slideHeight = verticalScale(390);
const slideWidth = wp(85);
const itemHorizontalMargin = wp(2);

export const sliderWidth = viewportWidth;
export const itemWidth = slideWidth + itemHorizontalMargin * 2;

const entryBorderRadius = verticalScale(10);

export default StyleSheet.create({
  slideInnerContainer: {
    width: itemWidth,
    height: slideHeight,
    shadowColor: colors.black,
    shadowOpacity: 0.16,
    shadowOffset: { width: 0, height: verticalScale(6) },
    shadowRadius: verticalScale(6),
    elevation: IS_IOS ? 1 : 5,
    backgroundColor: "white",
    borderRadius: entryBorderRadius,
  },
  shadow: {
    backgroundColor: IS_IOS ? "white" : "transparent",
    position: "absolute",
    top: 0,
    left: itemHorizontalMargin,
    right: itemHorizontalMargin,
    bottom: 18,
    shadowColor: colors.black,
    shadowOpacity: 0.16,
    shadowOffset: { width: 0, height: verticalScale(6) },
    shadowRadius: verticalScale(6),
    elevation: IS_IOS ? 1 : 5,
    borderRadius: entryBorderRadius,
    borderWidth: 0.4,
  },
  imageContainer: {
    flex: 1,
    backgroundColor: "white",
    borderTopLeftRadius: entryBorderRadius,
    borderTopRightRadius: entryBorderRadius,
  },
  imageContainerEven: {
    backgroundColor: colors.black,
  },
  image: {
    // ...StyleSheet.absoluteFillObject,
    flex: 1,
    resizeMode: "cover",
    borderRadius: IS_IOS ? entryBorderRadius : 0,
    borderTopLeftRadius: entryBorderRadius,
    borderTopRightRadius: entryBorderRadius,
    // borderBottomLeftRadius: 0,
    // borderBottomRightRadius: 0
    // borderRadius: 0
  },
  // image's border radius is buggy on iOS; let's hack it!
  radiusMask: {
    position: "absolute",
    bottom: 0,
    left: 0,
    right: 0,
    height: verticalScale(5),
    backgroundColor: "white",
  },
  radiusMaskEven: {
    backgroundColor: colors.black,
  },
  dealInfoContainer: {
    flexDirection: "row",
    justifyContent: "space-between",
    // paddingTop: 20 - entryBorderRadius,
    paddingTop: verticalScale(10) - entryBorderRadius,
    paddingBottom: verticalScale(15),
    paddingHorizontal: scale(15),
    backgroundColor: "white",
    borderBottomLeftRadius: entryBorderRadius,
    borderBottomRightRadius: entryBorderRadius,
  },
  dealLogoContainer: {
    flexDirection: "column",
    justifyContent: "flex-start",
    alignItems: "center",
  },
  dealLogoWrapper: {
    width: verticalScale(35),
    height: verticalScale(35),
    borderRadius: verticalScale(17.5),
    overflow: "hidden",
    alignItems: "center",
    justifyContent: "center",
  },
  dealLogo: {
    width: verticalScale(37),
    height: verticalScale(37),
    // resizeMode: "cover",
  },
  dealBranch: {
    marginTop: verticalScale(3),
    fontSize: moderateScale(10),
    width: 80,
    textAlign: 'center',
    color: "#6E6969",
  },
  dealTextContainer: {
    flex: 1,
    flexDirection: "column",
    paddingLeft: scale(15),
  },
  subtitle: {
    color: "#6E6969",
    fontSize: moderateScale(12),
  },
  features: {
    flexDirection: "row",
    alignItems: "flex-start",
    justifyContent: "flex-start",
  },
  featureIcons: {
    width: scale(10),
    height: verticalScale(10),
  },
  dealPerformanceContainer: {
    marginTop: verticalScale(10),
    flexDirection: "row",
  },
  likeContainer: {
    flexDirection: "row",
    alignItems: "flex-start",
    flex: 1,
  },
  likeIcon: {
    fontSize: moderateScale(12),
    color: "#5F92F3",
  },
  performanceText: {
    marginLeft: scale(5),
    fontSize: moderateScale(12),
    color: "#6E6969",
  },
  watchContainer: {
    flexDirection: "row",
    justifyContent: "flex-end",
    flex: 1,
  },
  watchIcon: {
    fontSize: moderateScale(7),
    color: "#000",
  },
});
