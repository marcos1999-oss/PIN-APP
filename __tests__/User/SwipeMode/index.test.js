/**
 * @format
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */

import 'react-native';
import React from 'react';
import configureMockStore from "redux-mock-store";
import SwipeMode from '../../../src/Screen/User/SwipeMode';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const mockStore = configureMockStore();

jest.mock('react-native-cached-images', () => {
  return {
    DocumentDir: () => {},
    ImageCache: {
      get: {
        clear: () => {},
      },
    },
  }
});

const store = mockStore({
  posts: {}
});

test('renders correctly', () => {
  const postsMeta = {
    loading: true,
  };
  const tree = renderer.create(<SwipeMode postsMeta={postsMeta} store={store} />).toJSON();
  expect(tree).toMatchSnapshot();
});
