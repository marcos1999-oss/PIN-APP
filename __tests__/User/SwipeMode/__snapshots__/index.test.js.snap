// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "paddingVertical": 60,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
          },
          Object {
            "alignItems": "center",
            "bottom": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "top": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 400,
              "width": 400,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#5F92F3",
              "borderRadius": 50,
              "height": 100,
              "justifyContent": "center",
              "width": 100,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "SignPainter",
                "fontSize": 42,
              }
            }
          >
            scopin
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 100,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#C4C4C4",
          "fontSize": 19,
          "textAlign": "center",
        }
      }
    >
      Currently no new deals
      

      pass by locations to see more deals
    </Text>
  </View>
</View>
`;
